assets/eip-####/contracts/TipToken.sol

```diff
@@ -0,0 +1,400 @@
+// SPDX-License-Identifier: Unlicensed
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/token/ERC1155/ERC1155.sol";
+import "@openzeppelin/contracts/token/ERC1155/IERC1155.sol";
+import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+import "./ITipToken.sol";
+
+/// @title Micropayments Standard for NFTs and Multi Tokens
+/// @author Jules Lai
+/// @notice An implementation for a standard interface for tip tokens that allows tipping to holders of NFTs and multi tokens
+/// @dev Allows tipping for ERC-721 and ERC-1155 multi tokens. Their holders receive the rewards of ERC20 compatible deposits
+contract TipToken is ITipToken, ERC20, Ownable {
```

Comment:
This is an implementation. Standards should not provide implementations, but interfaces.

Reply:
> This is an implementation. Standards should not provide implementations, but interfaces.

I removed the implementation and added in 'Reference Implementation' section of proposal a link to my public repo for example implementation and tests